File: binding/src/network/libqnetworkrequestfactory.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qnetworkrequestfactory.html
pub const qnetworkrequestfactory = struct {

    /// New constructs a new QNetworkRequestFactory object.
    pub fn New() QtC.QNetworkRequestFactory {
        return qtc.QNetworkRequestFactory_new();
    }


    /// New2 constructs a new QNetworkRequestFactory object.
    pub fn New2(baseUrl: ?*anyopaque) QtC.QNetworkRequestFactory {
        return qtc.QNetworkRequestFactory_new2(@ptrCast(baseUrl));
    }


    /// New3 constructs a new QNetworkRequestFactory object.
    pub fn New3(other: ?*anyopaque) QtC.QNetworkRequestFactory {
        return qtc.QNetworkRequestFactory_new3(@ptrCast(other));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn BaseUrl(self: ?*anyopaque, ) QtC.QUrl {
        return qtc.QNetworkRequestFactory_BaseUrl(@ptrCast(self));
    }

    pub fn SetBaseUrl(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetBaseUrl(@ptrCast(self), @ptrCast(url));
    }

    pub fn SslConfiguration(self: ?*anyopaque, ) QtC.QSslConfiguration {
        return qtc.QNetworkRequestFactory_SslConfiguration(@ptrCast(self));
    }

    pub fn SetSslConfiguration(self: ?*anyopaque, configuration: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetSslConfiguration(@ptrCast(self), @ptrCast(configuration));
    }

    pub fn CreateRequest(self: ?*anyopaque, ) QtC.QNetworkRequest {
        return qtc.QNetworkRequestFactory_CreateRequest(@ptrCast(self));
    }

    pub fn CreateRequest2(self: ?*anyopaque, query: ?*anyopaque) QtC.QNetworkRequest {
        return qtc.QNetworkRequestFactory_CreateRequest2(@ptrCast(self), @ptrCast(query));
    }

    pub fn CreateRequest3(self: ?*anyopaque, path: []const u8) QtC.QNetworkRequest {
        const path_str = qtc.libqt_string{
    .len = path.len,
    .data = path.ptr,
};
return qtc.QNetworkRequestFactory_CreateRequest3(@ptrCast(self), path_str);
    }

    pub fn CreateRequest4(self: ?*anyopaque, path: []const u8, query: ?*anyopaque) QtC.QNetworkRequest {
        const path_str = qtc.libqt_string{
    .len = path.len,
    .data = path.ptr,
};
return qtc.QNetworkRequestFactory_CreateRequest4(@ptrCast(self), path_str, @ptrCast(query));
    }

    pub fn SetCommonHeaders(self: ?*anyopaque, headers: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetCommonHeaders(@ptrCast(self), @ptrCast(headers));
    }

    pub fn CommonHeaders(self: ?*anyopaque, ) QtC.QHttpHeaders {
        return qtc.QNetworkRequestFactory_CommonHeaders(@ptrCast(self));
    }

    pub fn ClearCommonHeaders(self: ?*anyopaque, ) void {
        qtc.QNetworkRequestFactory_ClearCommonHeaders(@ptrCast(self));
    }

    pub fn BearerToken(self: ?*anyopaque, , allocator: std.mem.Allocator) []u8 {
        const _bytearray: qtc.libqt_string = qtc.QNetworkRequestFactory_BearerToken(@ptrCast(self));
defer qtc.libqt_string_free(&_bytearray);
const _ret = allocator.alloc(u8, _bytearray.len) catch @panic("qnetworkrequestfactory.BearerToken: Memory allocation failed");
@memcpy(_ret, _bytearray.data[0.._bytearray.len]);
return  _ret;
    }

    pub fn SetBearerToken(self: ?*anyopaque, token: []u8) void {
        const token_str = qtc.libqt_string{
    .len = token.len,
    .data = token.ptr,
};
qtc.QNetworkRequestFactory_SetBearerToken(@ptrCast(self), token_str);
    }

    pub fn ClearBearerToken(self: ?*anyopaque, ) void {
        qtc.QNetworkRequestFactory_ClearBearerToken(@ptrCast(self));
    }

    pub fn UserName(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkRequestFactory_UserName(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkrequestfactory.UserName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetUserName(self: ?*anyopaque, userName: []const u8) void {
        const userName_str = qtc.libqt_string{
    .len = userName.len,
    .data = userName.ptr,
};
qtc.QNetworkRequestFactory_SetUserName(@ptrCast(self), userName_str);
    }

    pub fn ClearUserName(self: ?*anyopaque, ) void {
        qtc.QNetworkRequestFactory_ClearUserName(@ptrCast(self));
    }

    pub fn Password(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkRequestFactory_Password(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkrequestfactory.Password: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetPassword(self: ?*anyopaque, password: []const u8) void {
        const password_str = qtc.libqt_string{
    .len = password.len,
    .data = password.ptr,
};
qtc.QNetworkRequestFactory_SetPassword(@ptrCast(self), password_str);
    }

    pub fn ClearPassword(self: ?*anyopaque, ) void {
        qtc.QNetworkRequestFactory_ClearPassword(@ptrCast(self));
    }

    pub fn QueryParameters(self: ?*anyopaque, ) QtC.QUrlQuery {
        return qtc.QNetworkRequestFactory_QueryParameters(@ptrCast(self));
    }

    pub fn SetQueryParameters(self: ?*anyopaque, query: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetQueryParameters(@ptrCast(self), @ptrCast(query));
    }

    pub fn ClearQueryParameters(self: ?*anyopaque, ) void {
        qtc.QNetworkRequestFactory_ClearQueryParameters(@ptrCast(self));
    }

    pub fn SetPriority(self: ?*anyopaque, priority: i32) void {
        qtc.QNetworkRequestFactory_SetPriority(@ptrCast(self), @intCast(priority));
    }

    pub fn Priority(self: ?*anyopaque, ) i32 {
        return qtc.QNetworkRequestFactory_Priority(@ptrCast(self));
    }

    pub fn Attribute(self: ?*anyopaque, attribute: i32) QtC.QVariant {
        return qtc.QNetworkRequestFactory_Attribute(@ptrCast(self), @intCast(attribute));
    }

    pub fn Attribute2(self: ?*anyopaque, attribute: i32, defaultValue: ?*anyopaque) QtC.QVariant {
        return qtc.QNetworkRequestFactory_Attribute2(@ptrCast(self), @intCast(attribute), @ptrCast(defaultValue));
    }

    pub fn SetAttribute(self: ?*anyopaque, attribute: i32, value: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetAttribute(@ptrCast(self), @intCast(attribute), @ptrCast(value));
    }

    pub fn ClearAttribute(self: ?*anyopaque, attribute: i32) void {
        qtc.QNetworkRequestFactory_ClearAttribute(@ptrCast(self), @intCast(attribute));
    }

    pub fn ClearAttributes(self: ?*anyopaque, ) void {
        qtc.QNetworkRequestFactory_ClearAttributes(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_Delete(@ptrCast(self));
    }
};

File: binding/src/network/libqnetworkrequestfactory.cpp
#include <QByteArray>
#include <QDebug>
#include <QHttpHeaders>
#include <QNetworkRequest>
#include <QNetworkRequestFactory>
#include <QSslConfiguration>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QUrl>
#include <QUrlQuery>
#include <QVariant>
#include <qnetworkrequestfactory.h>
#include "libqnetworkrequestfactory.h"
#include "libqnetworkrequestfactory.hxx"

QNetworkRequestFactory* QNetworkRequestFactory_new() {
	 return new QNetworkRequestFactory();
}

QNetworkRequestFactory* QNetworkRequestFactory_new2(const QUrl* baseUrl) {
	 return new QNetworkRequestFactory(*baseUrl);
}

QNetworkRequestFactory* QNetworkRequestFactory_new3(const QNetworkRequestFactory* other) {
	 return new QNetworkRequestFactory(*other);
}

void QNetworkRequestFactory_OperatorAssign(QNetworkRequestFactory* self, const QNetworkRequestFactory* other) {
	self->operator=(*other);
}

void QNetworkRequestFactory_Swap(QNetworkRequestFactory* self, QNetworkRequestFactory* other) {
	self->swap(*other);
}

QUrl* QNetworkRequestFactory_BaseUrl(const QNetworkRequestFactory* self) {
	return new QUrl(self->baseUrl());
}

void QNetworkRequestFactory_SetBaseUrl(QNetworkRequestFactory* self, const QUrl* url) {
	self->setBaseUrl(*url);
}

QSslConfiguration* QNetworkRequestFactory_SslConfiguration(const QNetworkRequestFactory* self) {
	return new QSslConfiguration(self->sslConfiguration());
}

void QNetworkRequestFactory_SetSslConfiguration(QNetworkRequestFactory* self, const QSslConfiguration* configuration) {
	self->setSslConfiguration(*configuration);
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest(const QNetworkRequestFactory* self) {
	return new QNetworkRequest(self->createRequest());
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest2(const QNetworkRequestFactory* self, const QUrlQuery* query) {
	return new QNetworkRequest(self->createRequest(*query));
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest3(const QNetworkRequestFactory* self, const libqt_string path) {
	return new QNetworkRequest(self->createRequest(QString::fromUtf8(path.data, path.len)));
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest4(const QNetworkRequestFactory* self, const libqt_string path, const QUrlQuery* query) {
	return new QNetworkRequest(self->createRequest(QString::fromUtf8(path.data, path.len), *query));
}

void QNetworkRequestFactory_SetCommonHeaders(QNetworkRequestFactory* self, const QHttpHeaders* headers) {
	self->setCommonHeaders(*headers);
}

QHttpHeaders* QNetworkRequestFactory_CommonHeaders(const QNetworkRequestFactory* self) {
	return new QHttpHeaders(self->commonHeaders());
}

void QNetworkRequestFactory_ClearCommonHeaders(QNetworkRequestFactory* self) {
	self->clearCommonHeaders();
}

libqt_string QNetworkRequestFactory_BearerToken(const QNetworkRequestFactory* self) {
	QByteArray _qb = self->bearerToken();
	libqt_string _str;
	_str.len = _qb.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _qb.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QNetworkRequestFactory_SetBearerToken(QNetworkRequestFactory* self, const libqt_string token) {
	self->setBearerToken(QByteArray(token.data, token.len));
}

void QNetworkRequestFactory_ClearBearerToken(QNetworkRequestFactory* self) {
	self->clearBearerToken();
}

libqt_string QNetworkRequestFactory_UserName(const QNetworkRequestFactory* self) {
	QString _ret = self->userName();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QNetworkRequestFactory_SetUserName(QNetworkRequestFactory* self, const libqt_string userName) {
	self->setUserName(QString::fromUtf8(userName.data, userName.len));
}

void QNetworkRequestFactory_ClearUserName(QNetworkRequestFactory* self) {
	self->clearUserName();
}

libqt_string QNetworkRequestFactory_Password(const QNetworkRequestFactory* self) {
	QString _ret = self->password();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QNetworkRequestFactory_SetPassword(QNetworkRequestFactory* self, const libqt_string password) {
	self->setPassword(QString::fromUtf8(password.data, password.len));
}

void QNetworkRequestFactory_ClearPassword(QNetworkRequestFactory* self) {
	self->clearPassword();
}

QUrlQuery* QNetworkRequestFactory_QueryParameters(const QNetworkRequestFactory* self) {
	return new QUrlQuery(self->queryParameters());
}

void QNetworkRequestFactory_SetQueryParameters(QNetworkRequestFactory* self, const QUrlQuery* query) {
	self->setQueryParameters(*query);
}

void QNetworkRequestFactory_ClearQueryParameters(QNetworkRequestFactory* self) {
	self->clearQueryParameters();
}

void QNetworkRequestFactory_SetPriority(QNetworkRequestFactory* self, int priority) {
	self->setPriority(static_cast<QNetworkRequest::Priority>(priority));
}

int QNetworkRequestFactory_Priority(const QNetworkRequestFactory* self) {
	return self->priority();
}

QVariant* QNetworkRequestFactory_Attribute(const QNetworkRequestFactory* self, int attribute) {
	return new QVariant(self->attribute(static_cast<QNetworkRequest::Attribute>(attribute)));
}

QVariant* QNetworkRequestFactory_Attribute2(const QNetworkRequestFactory* self, int attribute, const QVariant* defaultValue) {
	return new QVariant(self->attribute(static_cast<QNetworkRequest::Attribute>(attribute), *defaultValue));
}

void QNetworkRequestFactory_SetAttribute(QNetworkRequestFactory* self, int attribute, const QVariant* value) {
	self->setAttribute(static_cast<QNetworkRequest::Attribute>(attribute), *value);
}

void QNetworkRequestFactory_ClearAttribute(QNetworkRequestFactory* self, int attribute) {
	self->clearAttribute(static_cast<QNetworkRequest::Attribute>(attribute));
}

void QNetworkRequestFactory_ClearAttributes(QNetworkRequestFactory* self) {
	self->clearAttributes();
}

void QNetworkRequestFactory_Delete(QNetworkRequestFactory* self) {
    delete self;
}

